src/ReactionNetwork.jl

```diff
@@ -0,0 +1,341 @@
+    params = collect(keys(parameters))
+
+    f_expr = get_f(reactions, reactants)
+    f = make_func(f_expr, reactants, parameters)
+
+    g_expr = get_g(reactions, reactants)
+    g = make_func(g_expr, reactants, parameters)
+    p_matrix = zeros(length(reactants), length(reactions))
+
+    (jump_rate_expr, jump_affect_expr) = get_jump_expr(reactions, reactants)
+    jumps = get_jumps(jump_rate_expr, jump_affect_expr,reactants,reactants)
+
+    f_rhs = [element.args[2] for element in f_expr]
+    symjac = Expr(:quote, calculate_jac(f_rhs, syms))
+    f_symfuncs = hcat([SymEngine.Basic(f) for f in f_rhs])
+
+    # Build the type
+    exprs = Vector{Expr}(0)
+
+    ## only get the right-hand-side of the equations.
+    f_funcs = [element.args[2] for element in f_expr]
+    g_funcs = [element.args[2] for element in g_expr]
+
+    typeex,constructorex = maketype(name, f, f_funcs, f_symfuncs, g, g_funcs, jumps, Meta.quot(jump_rate_expr), Meta.quot(jump_affect_expr), p_matrix, syms; params=params, symjac=symjac)
+    push!(exprs,typeex)
+    push!(exprs,constructorex)
+
+    # export type constructor
+    def_const_ex = :(($name)()) |> esc
+    push!(exprs,def_const_ex)
+
+    expr_arr_to_block(exprs)
+end
+
+#Generates a vector containing a number of reaction structures, each containing the infromation about one reaction.
+function get_reactions(ex::Expr)
+    reactions = Vector{ReactionStruct}(0)      ::Vector{ReactionStruct}
+    for line in ex.args
+        (line.head != :tuple) && (continue)
+        (rate,r_line) = line.args
+
+        if r_line.head  == :-->
+            r_line = Expr(:call,:→,r_line.args[1],r_line.args[2])
+        end
+
+        arrow = r_line.args[1]  ::Symbol
+        if in(arrow,double_arrows)
+            push_reactions(reactions::Vector{ReactionStruct}, r_line.args[2], r_line.args[3], rate.args[1], !in(arrow,no_mass_arrows))
+            push_reactions(reactions::Vector{ReactionStruct}, r_line.args[3], r_line.args[2], rate.args[2], !in(arrow,no_mass_arrows))
+        elseif in(arrow,fwd_arrows)
+            push_reactions(reactions::Vector{ReactionStruct}, r_line.args[2], r_line.args[3], rate, !in(arrow,no_mass_arrows))
+        elseif in(arrow,bwd_arrows)
+            push_reactions(reactions::Vector{ReactionStruct}, r_line.args[3], r_line.args[2], rate, !in(arrow,no_mass_arrows))
+        else
+            throw("malformed reaction")
+        end
+    end
+    return reactions
+end
+
+#Structure containing information about one reactant in one reaction.
+struct ReactantStruct
+    reactant::Symbol
+    stoichiometry::Int64
+end
+
+#Structure containing information about one Reaction. Contain all its substrates and products as well as its rate. Contains an specialized constructor.
+struct ReactionStruct
+    substrates::Vector{ReactantStruct}
+    products::Vector{ReactantStruct}
+    rate::Any
+    function ReactionStruct(sub_line::Any, prod_line::Any, rate::Any, use_mass_kin::Bool)
```

Comment:
there's no need to `::Any`. That's the same thing as leaving it off. Just a style thing.